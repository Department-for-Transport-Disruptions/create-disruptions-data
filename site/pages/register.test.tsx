import renderer from "react-test-renderer";
import { describe, it, expect } from "vitest";
import Register, { RegisterPageProps } from "./register.page";

const blankInputs: RegisterPageProps = {
    errors: [],
    inputs: {},
};

const withInputsAndNoErrors: RegisterPageProps = {
    errors: [],
    inputs: {
        email: "dummyUser@gmail.com",
        key: "key123",
        password: "dummyPassword",
        confirmPassword: "dummyPassword",
        organisation: "Test Org",
    },
};

const withInputsAndErrors: RegisterPageProps = {
    ...withInputsAndNoErrors,
    errors: [{ errorMessage: "Enter a valid email address", id: "email" }],
};

describe("register", () => {
    it("should render correctly when there are no inputs", () => {
        const tree = renderer.create(<Register {...blankInputs} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render correctly with inputs and no errors", () => {
        const tree = renderer.create(<Register {...withInputsAndNoErrors} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render correctly with inputs and with errors", () => {
        const tree = renderer.create(<Register {...withInputsAndErrors} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
