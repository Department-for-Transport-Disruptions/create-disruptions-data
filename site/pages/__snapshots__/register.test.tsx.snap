// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`register > should render correctly when there are no inputs 1`] = `
[
  <div
    className="app-phase-banner__wrapper"
  >
    <div
      className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        className="govuk-phase-banner__content"
      >
        <strong
          className="govuk-tag govuk-phase-banner__content__tag"
        >
          alpha
        </strong>
        <span
          className="govuk-phase-banner__text"
        >
          This is a new service â€“ your
           
          <a
            className="govuk-link"
            href="/feedback"
            id="feedback-link"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            feedback
          </a>
           
          will help us to improve it.
        </span>
      </p>
    </div>
  </div>,
  <div
    className="govuk-width-container"
  >
    <main
      className="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        className="govuk-heading-xl"
      >
        Create an account
      </h1>
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/register"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <table
              className="govuk-table"
            >
              <thead
                className="govuk-table__head"
              >
                <tr
                  className="govuk-table__row"
                />
              </thead>
              <tbody
                className="govuk-table__body"
              >
                <tr
                  className="govuk-table__row"
                >
                  <th
                    className="govuk-table__header align-middle"
                    scope="row"
                  >
                    Email address
                  </th>
                  <td
                    className="govuk-table__cell align-middle"
                  />
                </tr>
              </tbody>
            </table>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
                id="password"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="password-input"
                >
                  Password
                </label>
                <div
                  className="govuk-hint"
                  id="password-hint"
                >
                  Your password should be at least 8 characters long
                </div>
                <div
                  className=""
                >
                  <input
                    aria-describedby="password-hint"
                    className="govuk-input w-3/4"
                    id="password-input"
                    maxLength={100}
                    name="password"
                    onBlur={[Function]}
                    type="password"
                  />
                </div>
              </div>
            </div>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
                id="confirm-password"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="confirm-password-input"
                >
                  Confirm password
                </label>
                <div
                  className=""
                >
                  <input
                    className="govuk-input w-3/4"
                    id="confirm-password-input"
                    maxLength={100}
                    name="confirmPassword"
                    onBlur={[Function]}
                    type="password"
                  />
                </div>
              </div>
            </div>
            <input
              name="email"
              type="hidden"
            />
            <input
              name="key"
              type="hidden"
            />
            <button
              className="govuk-button mt-8"
            >
              Save password
            </button>
          </form>
        </div>
      </div>
    </main>
    <div>
      <h2
        className="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        className="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link:
         
        <a
          className="govuk-link govuk-!-font-size-19"
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Contact us
        </a>
      </p>
    </div>
  </div>,
  <footer
    className="govuk-footer"
    role="contentinfo"
  >
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-footer__meta"
      >
        <div
          className="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            className="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            className="govuk-footer__inline-list"
          >
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Help
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/cookies"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Cookies
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Contact
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/accessibility"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Accessibility
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Terms and conditions
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/privacy"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            className="govuk-footer__body"
          >
            Built by the
             
            <a
              className="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            className="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            className="govuk-footer__licence-description"
          >
            All content is available under the
             
            <a
              className="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          className="govuk-footer__meta-item"
        >
          <a
            className="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;

exports[`register > should render correctly with inputs and no errors 1`] = `
[
  <div
    className="app-phase-banner__wrapper"
  >
    <div
      className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        className="govuk-phase-banner__content"
      >
        <strong
          className="govuk-tag govuk-phase-banner__content__tag"
        >
          alpha
        </strong>
        <span
          className="govuk-phase-banner__text"
        >
          This is a new service â€“ your
           
          <a
            className="govuk-link"
            href="/feedback"
            id="feedback-link"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            feedback
          </a>
           
          will help us to improve it.
        </span>
      </p>
    </div>
  </div>,
  <div
    className="govuk-width-container"
  >
    <main
      className="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        className="govuk-heading-xl"
      >
        Create an account
      </h1>
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/register"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <table
              className="govuk-table"
            >
              <thead
                className="govuk-table__head"
              >
                <tr
                  className="govuk-table__row"
                />
              </thead>
              <tbody
                className="govuk-table__body"
              >
                <tr
                  className="govuk-table__row"
                >
                  <th
                    className="govuk-table__header align-middle"
                    scope="row"
                  >
                    Email address
                  </th>
                  <td
                    className="govuk-table__cell align-middle"
                  >
                    dummyUser@gmail.com
                  </td>
                </tr>
                <tr
                  className="govuk-table__row"
                >
                  <th
                    className="govuk-table__header align-middle"
                    scope="row"
                  >
                    Organisation
                  </th>
                  <td
                    className="govuk-table__cell align-middle"
                  >
                    Test Org
                  </td>
                </tr>
              </tbody>
            </table>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
                id="password"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="password-input"
                >
                  Password
                </label>
                <div
                  className="govuk-hint"
                  id="password-hint"
                >
                  Your password should be at least 8 characters long
                </div>
                <div
                  className=""
                >
                  <input
                    aria-describedby="password-hint"
                    className="govuk-input w-3/4"
                    defaultValue="dummyPassword"
                    id="password-input"
                    maxLength={100}
                    name="password"
                    onBlur={[Function]}
                    type="password"
                  />
                </div>
              </div>
            </div>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
                id="confirm-password"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="confirm-password-input"
                >
                  Confirm password
                </label>
                <div
                  className=""
                >
                  <input
                    className="govuk-input w-3/4"
                    defaultValue="dummyPassword"
                    id="confirm-password-input"
                    maxLength={100}
                    name="confirmPassword"
                    onBlur={[Function]}
                    type="password"
                  />
                </div>
              </div>
            </div>
            <input
              name="email"
              type="hidden"
              value="dummyUser@gmail.com"
            />
            <input
              name="key"
              type="hidden"
              value="key123"
            />
            <button
              className="govuk-button mt-8"
            >
              Save password
            </button>
          </form>
        </div>
      </div>
    </main>
    <div>
      <h2
        className="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        className="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link:
         
        <a
          className="govuk-link govuk-!-font-size-19"
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Contact us
        </a>
      </p>
    </div>
  </div>,
  <footer
    className="govuk-footer"
    role="contentinfo"
  >
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-footer__meta"
      >
        <div
          className="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            className="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            className="govuk-footer__inline-list"
          >
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Help
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/cookies"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Cookies
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Contact
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/accessibility"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Accessibility
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Terms and conditions
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/privacy"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            className="govuk-footer__body"
          >
            Built by the
             
            <a
              className="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            className="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            className="govuk-footer__licence-description"
          >
            All content is available under the
             
            <a
              className="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          className="govuk-footer__meta-item"
        >
          <a
            className="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;

exports[`register > should render correctly with inputs and with errors 1`] = `
[
  <div
    className="app-phase-banner__wrapper"
  >
    <div
      className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        className="govuk-phase-banner__content"
      >
        <strong
          className="govuk-tag govuk-phase-banner__content__tag"
        >
          alpha
        </strong>
        <span
          className="govuk-phase-banner__text"
        >
          This is a new service â€“ your
           
          <a
            className="govuk-link"
            href="/feedback"
            id="feedback-link"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            feedback
          </a>
           
          will help us to improve it.
        </span>
      </p>
    </div>
  </div>,
  <div
    className="govuk-width-container"
  >
    <main
      className="govuk-main-wrapper"
      id="main-content"
    >
      <h1
        className="govuk-heading-xl"
      >
        Create an account
      </h1>
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/register"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <div
              aria-labelledby="error-summary-title"
              className="govuk-error-summary"
              data-module="govuk-error-summary"
              role="alert"
              tabIndex={-1}
            >
              <h2
                className="govuk-error-summary__title"
                id="error-summary-title"
              >
                There is a problem
              </h2>
              <div
                className="govuk-error-summary__body"
              >
                <ul
                  className="govuk-list govuk-error-summary__list"
                >
                  <li>
                    <a
                      href="#email"
                    >
                      Enter a valid email address
                    </a>
                  </li>
                </ul>
              </div>
            </div>
            <table
              className="govuk-table"
            >
              <thead
                className="govuk-table__head"
              >
                <tr
                  className="govuk-table__row"
                />
              </thead>
              <tbody
                className="govuk-table__body"
              >
                <tr
                  className="govuk-table__row"
                >
                  <th
                    className="govuk-table__header align-middle"
                    scope="row"
                  >
                    Email address
                  </th>
                  <td
                    className="govuk-table__cell align-middle"
                  >
                    dummyUser@gmail.com
                  </td>
                </tr>
                <tr
                  className="govuk-table__row"
                >
                  <th
                    className="govuk-table__header align-middle"
                    scope="row"
                  >
                    Organisation
                  </th>
                  <td
                    className="govuk-table__cell align-middle"
                  >
                    Test Org
                  </td>
                </tr>
              </tbody>
            </table>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
                id="password"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="password-input"
                >
                  Password
                </label>
                <div
                  className="govuk-hint"
                  id="password-hint"
                >
                  Your password should be at least 8 characters long
                </div>
                <div
                  className=""
                >
                  <input
                    aria-describedby="password-hint"
                    className="govuk-input w-3/4"
                    defaultValue="dummyPassword"
                    id="password-input"
                    maxLength={100}
                    name="password"
                    onBlur={[Function]}
                    type="password"
                  />
                </div>
              </div>
            </div>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
                id="confirm-password"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="confirm-password-input"
                >
                  Confirm password
                </label>
                <div
                  className=""
                >
                  <input
                    className="govuk-input w-3/4"
                    defaultValue="dummyPassword"
                    id="confirm-password-input"
                    maxLength={100}
                    name="confirmPassword"
                    onBlur={[Function]}
                    type="password"
                  />
                </div>
              </div>
            </div>
            <input
              name="email"
              type="hidden"
              value="dummyUser@gmail.com"
            />
            <input
              name="key"
              type="hidden"
              value="key123"
            />
            <button
              className="govuk-button mt-8"
            >
              Save password
            </button>
          </form>
        </div>
      </div>
    </main>
    <div>
      <h2
        className="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        className="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link:
         
        <a
          className="govuk-link govuk-!-font-size-19"
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Contact us
        </a>
      </p>
    </div>
  </div>,
  <footer
    className="govuk-footer"
    role="contentinfo"
  >
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-footer__meta"
      >
        <div
          className="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            className="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            className="govuk-footer__inline-list"
          >
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Help
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/cookies"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Cookies
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Contact
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/accessibility"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Accessibility
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Terms and conditions
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/privacy"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            className="govuk-footer__body"
          >
            Built by the
             
            <a
              className="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            className="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            className="govuk-footer__licence-description"
          >
            All content is available under the
             
            <a
              className="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          className="govuk-footer__meta-item"
        >
          <a
            className="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;
